source/vibe/db/postgresql/package.d

```diff
@@ -107,16 +107,32 @@ class __Conn : dpq2.Connection
     {
         import vibe.core.core;
 
+        // Connection socket should be non-blocking while waiting
         auto sock = this.socket();
-
         sock.blocking = false;
         scope(exit)
         {
             sock.blocking = true;
             destroy(sock);
         }
 
-        auto event = createFileDescriptorEvent(sock.handle, FileDescriptorEvent.Trigger.any);
+        version(Have_vibe_core)
+        {
+            // vibe-core right now supports only read trigger event
+            enum trigger = FileDescriptorEvent.Trigger.read;
+
+            // vibe-core does reference counting and may close socket
+            // after it are used.
+            // Thus we need to a copy of socket for it.
+            auto posix_socket = cast(int) sock.posixSocketDuplicate;
```

Comment:
There is no need to use sock, we could just use this here

Reply:
(See above)

Reply:
BTW: sock does not have posixSocketDuplicate method

Reply:
Fixed. It is need to add Have_vibe_core version into CI